commitizen/commands/init.py

```diff
@@ -113,194 +110,176 @@ def __call__(self):
+                choices=list(registry.keys()),
+                default="cz_conventional_commits",
+                style=self.cz.style,
+            ).unsafe_ask()
+        )
 
     def _ask_tag(self) -> str:
-        latest_tag = self.project_info.latest_tag
+        latest_tag = get_latest_tag_name()
         if not latest_tag:
             out.error("No Existing Tag. Set tag to v0.0.1")
             return "0.0.1"
 
         is_correct_tag = questionary.confirm(
             f"Is {latest_tag} the latest tag?", style=self.cz.style, default=False
         ).unsafe_ask()
-        if not is_correct_tag:
-            tags = self.project_info.tags()
-            if not tags:
-                out.error("No Existing Tag. Set tag to v0.0.1")
-                return "0.0.1"
-
-            # the latest tag is most likely with the largest number. Thus list the tags in reverse order makes more sense
-            sorted_tags = sorted(tags, reverse=True)
-            latest_tag = questionary.select(
+        if is_correct_tag:
+            return latest_tag
+
+        tags = get_tag_names()
+        if not tags:
+            out.error("No Existing Tag. Set tag to v0.0.1")
+            return "0.0.1"
+
+        # Tags are sorted in reverse order to get the most recent tag first,
+        # which is typically the one we want to use as a reference.
+        latest_tag = str(
+            questionary.select(
                 "Please choose the latest tag: ",
-                choices=sorted_tags,
+                choices=sorted(tags, reverse=True),
                 style=self.cz.style,
             ).unsafe_ask()
+        )
+
+        if not latest_tag:
+            raise NoAnswersError("Tag is required!")
 
-            if not latest_tag:
-                raise NoAnswersError("Tag is required!")
         return latest_tag
 
     def _ask_tag_format(self, latest_tag) -> str:
-        is_correct_format = False
         if latest_tag.startswith("v"):
             tag_format = r"v$version"
             is_correct_format = questionary.confirm(
                 f'Is "{tag_format}" the correct tag format?', style=self.cz.style
             ).unsafe_ask()
+            if is_correct_format:
+                return tag_format
 
         default_format = DEFAULT_SETTINGS["tag_format"]
-        if not is_correct_format:
-            tag_format = questionary.text(
-                f'Please enter the correct version format: (default: "{default_format}")',
-                style=self.cz.style,
-            ).unsafe_ask()
+        tag_format = questionary.text(
+            f'Please enter the correct version format: (default: "{default_format}")',
+            style=self.cz.style,
+        ).unsafe_ask()
 
-            if not tag_format:
-                tag_format = default_format
-        return tag_format
+        return tag_format or default_format
 
     def _ask_version_provider(self) -> str:
         """Ask for setting: version_provider"""
 
         OPTS = {
```

Comment:
Ideally, the `VersionProvider` ABC class should include `identifier` and `description`. Which would allow us to do something like

```sh
providers = metadata.entry_points(group=PROVIDER_ENTRYPOINT)

{provider.identifier: provider.description for provider in providers} 
```

This way, custom user made providers would also appear in the init functions.

---

Just to be clear, this is optional, I hardcoded them here because it would help 90% of the users and I didn't want to go further.